conda_smithy/ci_register.py

```diff
@@ -192,17 +192,36 @@ def add_project_to_travis(user, project):
             if "repo" in content:
                 break
         except requests.HTTPError as e:
-            # We regularly seem to hit this issue during automated feedstock registration on github.
-            # https://github.com/conda-forge/conda-smithy/issues/233
-            # Repo is not available on travis-ci yet.
             print(e)
 
         if count == 1:
+            def wait_until_synced(ignore=False):
+                is_sync_url = '{}/users'.format(endpoint)
+                response = requests.post(is_sync_url, headers=headers)
+                content = response.json()
+                for c in range(20):
+                    if ("is_syncing" in content and content["is_syncing"] == False):
+                        break
+                    time.sleep(3)
+                else:
+                    if ignore:
+                        print(" * Travis is being synced by somebody else. Ignoring")
+                    else:
+                        raise RuntimeError('Synching is not finishing for a minute now')
+
+            wait_until_synced(ignore=True)
```

Comment:
To put it simply, we should check after the currently running sync has run to see if this repo was added. Only if it was not added, should we try to sync ourselves.